closed/make/Docs.gmk

```diff
@@ -21,3 +21,103 @@
+
+JAVADOC_DOCROOT := \(\.\./\)*\.\.
+JAVADOC_BOTTOM_REGEX = $(subst {@docroot},$(JAVADOC_DOCROOT),$(subst .,\.,$(JAVADOC_BOTTOM)))
+OPENJ9_JAVADOC_BOTTOM := $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
+OPENJ9_SUBST_COMMAND = $(SED) -e '\''s|$(JAVADOC_BOTTOM_REGEX)|$(OPENJ9_JAVADOC_BOTTOM)|g'\''
+
+################################################################################
+# Define a javadoc Group for the OpenJ9 specific modules
+# Module names are added to DOCS_MODULES in closed/make/common/Modules.gmk
+OPENJ9_GROUP_NAME := OpenJ9
+OPENJ9_GROUP_MODULES := openj9.*
+OPENJ9_GROUP_DESCRIPTION := \
+    Modules whose names start with {@code openj9.} contain APIs provided by the \
+    <a href="$(OPENJ9_JAVADOC_BASE_URL)">Eclipse OpenJ9</a> project. \
+    These APIs can only be used with a JDK which includes the OpenJ9 virtual machine. \
+    #
+JDK_GROUPS += OPENJ9
+
+################################################################################
+# Add openj9 specific modules to the list of modules for which javadoc is required
+OPENJ9_MODULES := $(filter openj9.%,$(call FindAllModules))
+DOCS_MODULES += $(OPENJ9_MODULES)
+
+OPENJDK_JAVADOC_MARKER := $(SUPPORT_OUTPUTDIR)/docs/_javadoc_JDK_API_exec.marker
+OPENJ9_JAVADOC_MARKER := $(SUPPORT_OUTPUTDIR)/docs/_javadoc_JDK_API_openj9.marker
+JDK_API_CUSTOM_TARGETS += $(OPENJ9_JAVADOC_MARKER)
+
+OPENJ9_JAVADOC_SCRIPT := $(SUPPORT_OUTPUTDIR)/docs/_javadoc_openj9_footer_replace.sh
+OPENJ9_JAVADOC_TEMP  := $(SUPPORT_OUTPUTDIR)/docs/_javadoc_temp.html
+
+################################################################################
+# The OpenJ9 classes are built from the j9jcl_sources and closed/src
+# directories.
+# The packages are beneath directories called /share/classes (or /unix/classes,
+# /windows/classes etc.) which do not appear in the javadoc tree structure.
+# The commands below replace the footer in the javadoc html pages for the
+# following files:
+# 1. The openj9 java source files which override the openjdk versions
+# 2. The openj9.xxxx modules, including the javadoc generated files such as
+#    enum class pages, module-summary.html, package-summary.html etc.)
+# 3. The com/ibm/xxxx packages, including the javadoc generated files such as
+#    enum class pages, module-summary.html, package-summary.html etc.)
+# A temporary script file is used to avoid xargs command line length issues
+# on some platforms. 
+
+OPENJ9_JAVADOC_SCRIPT_BODY = \
+    for path in "$$@" ; do \
+      if [ -f "$(DOCS_OUTPUTDIR)/api/$$path" ] ; then \
+        $(CP) "$(DOCS_OUTPUTDIR)/api/$$path" "$(OPENJ9_JAVADOC_TEMP)" ; \
+        $(OPENJ9_SUBST_COMMAND) < "$(OPENJ9_JAVADOC_TEMP)" > "$(DOCS_OUTPUTDIR)/api/$$path" ; \
+        $(RM) "$(OPENJ9_JAVADOC_TEMP)" ; \
+      fi ; \
+    done \
+    #
+
+# Dependency on the openjdk marker file ensures the footers are only
+# replaced after the javadoc has been created.
+# on some platforms. 
+$(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
```

Comment:
This is different than jdk11: does this make sense to apply there?

nit: Please remove trailing whitespace from all lines.

Reply:
I found 2 trailing spaces.  Did you see others?
I also removed the 'on some platforms' line which shouldn't be there.